### PR TITLE
Migrate Channelstore categories to sqlx

### PR DESCRIPTION
**Pull request by agnivade**
Remove db:"-" tags from model.Channel as well.

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/channel_store_categories.go [hunk 1]
```diff
@@ -436,7 +447,7 @@ func (s SqlChannelStore) GetSidebarCategory(categoryId string) (*model.SidebarCa
 	}
 
 	if _, err = s.GetReplica().Select(&categories, sql, args...); err != nil {
-		return nil, store.NewErrNotFound("SidebarCategories", categoryId)
+		return nil, errors.Wrap(err, "hello error")
```
Comment (isacikgoz) [inline store/sqlstore/channel_store_categories.go:450]: What is "hello error"? 😅

File: store/sqlstore/channel_store_categories.go [hunk 2]
```diff
@@ -475,8 +486,8 @@ func (s SqlChannelStore) getSidebarCategoriesT(db dbSelecter, userId, teamId str
 		return nil, errors.Wrap(err, "sidebar_categories_tosql")
 	}
 
-	if _, err = db.Select(&categories, query, args...); err != nil {
-		return nil, store.NewErrNotFound("SidebarCategories", fmt.Sprintf("userId=%s,teamId=%s", userId, teamId))
+	if err := db.Select(&categories, query, args...); err != nil {
+		return nil, errors.Wrap(err, "hello error")
```
Comment (isacikgoz) [inline store/sqlstore/channel_store_categories.go:490]: Same here.

**Review by isacikgoz — COMMENTED** [review]
Looks good overall, have a question tho 👇

File: store/sqlstore/channel_store_categories.go (hunk 1 above)
Comment (agnivade) [reply store/sqlstore/channel_store_categories.go:450]: It is called a "left-during-debugging" error :rofl: Fixing.

**Review by isacikgoz — APPROVED** [review]
Awesome 💯

**Pull request merged**